cordova-lib/src/cordova/restore-util.js

```diff
@@ -64,7 +64,25 @@ function installPlatformsFromConfigXML(platforms, opts) {
             // Combining arrays and checking duplicates.
             comboArray = pkgJsonPlatforms.slice();
         }
+
         engines = cfg.getEngines(projectHome);
+
+        // TODO: CB-12592: Eventually refactor out to pacakge manager module.
+        // If package.json doesn't exist, auto-create one.
+        if (engines.length > 0 && !fs.existsSync(pkgJsonPath)) {
```

Comment:
oo i think it should have been `(engines.length > 0 && pkgJson === undefined)`. Does that work?